src/Connectors/src/Connectors/MongoDb/MongoDbConnectionStringBuilder.cs

```diff
@@ -0,0 +1,222 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Specialized;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Web;
+using Microsoft.AspNetCore.Http;
+using Steeltoe.Common;
+
+namespace Steeltoe.Connectors.MongoDb;
+
+/// <summary>
+/// Provides a connection string builder for MongoDB URLs. The format is defined at: https://www.mongodb.com/docs/manual/reference/connection-string/.
+/// </summary>
+internal sealed class MongoDbConnectionStringBuilder : IConnectionStringBuilder
+{
+    private readonly Dictionary<string, string> _settings = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
I was going to note that it seems as though we might get into a case sensitivity issue later in this file with the unknown keywords, but it looks like you took care of that in the initialization already, nice!